app/controllers/users_controller.rb

```diff
@@ -91,7 +91,8 @@ def admin_update_permissions
 
   def update_email_preferences
     prefs = params[:prefs]
-    authorize current_user, :update?
+    # Only allow the user to update their own preferences unless it is a super admin
+    raise Pundit::NotAuthorizedError unless current_user.can_super_admin? || current_user.id != params[:id]
```

Comment:
This line of code does not really fix the problem since unlike the above comment states, it does the following:
raise Pundit::NotAuthorizedError:
- when the current_user is super_admin or 
- when current_user id is different. 

I don't think we want to prevent a super_admin to update her email preferences neither it is needed to rely on the id passed at all since at the end we are using current_user to update the preferences.